codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -351,44 +378,28 @@ let registerDynamicContracts = (
   )
 }
 
-type partitionQuery = {
-  partitionId: string,
-  fromBlock: int,
-  toBlock: option<int>,
-  contractAddressMapping: ContractAddressingMap.mapping,
-}
+type queryTarget =
+  | Head
+  | EndBlock({toBlock: int})
+  | Merge({
+      // The partition we are going to merge into
+      // It shouldn't be fetching during the query
+      intoPartitionId: string,
+      toBlock: int,
+    })
+
+// Strip internal fields from partition kind like dynamicContracts
+type querySelection =
+  | Wildcard
+  | Normal({contractAddressMapping: ContractAddressingMap.mapping})
 
-type mergeQuery = {
-  // The catching up partition
+type query = {
   partitionId: string,
-  // The partition we are going to merge into
-  // It shouldn't be fetching during the query
-  intoPartitionId: string,
   fromBlock: int,
-  toBlock: int,
-  contractAddressMapping: ContractAddressingMap.mapping,
-}
-
-type query =
-  | PartitionQuery(partitionQuery)
-  | MergeQuery(mergeQuery)
-
-let queryFromBlock = query => {
-  switch query {
-  | PartitionQuery({fromBlock}) => fromBlock
-  | MergeQuery({fromBlock}) => fromBlock
-  }
-}
-
-let queryPartitionId = query => {
-  switch query {
-  | PartitionQuery({partitionId}) => partitionId
-  | MergeQuery({partitionId}) => partitionId
-  }
+  selection: querySelection,
+  target: queryTarget,
```

Comment:
Change to make it easier to work according to the suggestion https://github.com/enviodev/hyperindex/pull/405#discussion_r1904357883